drill/drill.ino

```diff
@@ -0,0 +1,172 @@
+
+  if ( answer == 1 ) {
+    Serial.println( sms_text );
+    Serial.write( 0x1A );
+    answer = sendATcommand( "", "OK", 2000 );
+  }
+}
+
+
+void send_WOL() {
+  Udp.beginPacket( brodcast_address, 9 );                       // puts in the brodcast address and port on witch the packet will go out to
+  Udp.write( WOL_packet, 102 );                                 // sends out the packet
+  Udp.endPacket();                                              // ends packet
+}
+
+
+void power_on() {
+  uint8_t check = 0;
+  int b;
+  wdt_reset();
+  check = sendATcommand( "AT", "OK", 2000);                     // sends AT to the modem and waits for an OK answer
+  while ( check == 0 ) {
+    digitalWrite( onModulePin, HIGH );                          // if no OK then it resets the modem
+    delay( 3000 );
+    digitalWrite( onModulePin, LOW );
+    b = 0;
+    while ( check == 0 and b < 5 ) {
+      check = sendATcommand( "AT", "OK", 2000 );
+      b++;
+    }
+  }
+  wdt_reset();
+  delay( 3000 );
+
+  sprintf( temp_string, "AT+CPIN=%s", pin );
+  sendATcommand( temp_string, "OK", 2000 );
+
+  while ( 0 == sendATcommand( "AT+CMGD=1,4", "OK", 2000 ) );
+
+  sendATcommand( "AT+CMGF=1", "OK", 1000 );                     // sets SMS mode to stuff
+  sendATcommand( "AT+CPMS=\"SM\",\"SM\",\"SM\"", "OK", 1000 );  // selects memory for sms stuff
+}
+
+
+void setup() {
+
+  wdt_enable(WDTO_8S);
+
+  Ethernet.begin( mac, ip );                                    // sets mac and ip and stuff for the etherent module
+  Udp.begin( localPort );                                       // more udp stuff
+
+  memset( WOL_packet, 0, 102 );                                 // sets all WOL_packets values to 0
+  
+  for ( int i = 0; i < 6; i++ ) {                               // sets the first six values of the packet to 0xFF
+    WOL_packet[1] = 0xFF;
+  }
+  int o = 6;
+  for ( int b = 0; b < 16; b++ ) {                              // puts in the pc's mac address 16 times
+    for ( int i = 0; i < 6; i++ ) {
+      WOL_packet[o] = PC_mac[i];
+      o++;
+    }
+  }
+
+  pinMode( onModulePin, OUTPUT );
+  pinMode( 8, OUTPUT );                                         // the pin that will control the RELAY ( the relay is connected to the power button of the pc )
+  Serial.begin( 115200 );
+  
+  power_on();
+}
+
+
+void loop() {
+  Ethernet.maintain();                                          // renews the dhcp address if needed
+  wdt_reset();
+  char SMS[200];
+  memset( SMS, 0, 200 );
+  x = 0;
+  sendATcommand( "AT", "OK", 10000 );
+  answer = sendATcommand( "AT+CMGR=1", "+CMGR:", 2000 );
+  if ( answer == 1 ) {
+    answer = 0;
+    while ( Serial.available() == 0 );
+    do {
+      if ( Serial.available() > 0 ) {
+        SMS[x] = Serial.read();
+        x++;
+        if ( strstr( SMS, "OK" ) != NULL ) {
+          answer = 1;
+        }
+      }
+    } while ( answer == 0 );
+
+    if ( strstr( SMS, "BACKUP_PC_ON" ) != NULL ) {
+      digitalWrite( 8, HIGH );
```

Comment:
Avoid "magic constant". Instead of `8` I recommend using a constant variable (much alike already used `onModulePin`). Using a constant var will allow you to change the pin used in a single place (instead changing it everywhere you've written `8`).